driver-core/src/main/java/com/datastax/driver/core/QueryLogger.java

```diff
@@ -647,7 +651,9 @@ protected void logQuery(Statement statement, Exception exception, Logger logger,
         boolean showParameterValues = logger.isTraceEnabled();
```

Comment:
I was wondering if we could extract `logQuery` and its dependent methods to another place and make it public (maybe `RegularStatement.toString()`? Or some helper class?). I can imagine users benefitting from this when they need to log statements in their own code. Logging parameters manually is tedious and can be dangerous if there are too many of them etc. and this method takes care of all this. E.g. in pull request #270 (JAVA-115) the author tries to do exactly that, but without all the guards and checks we have here.